src/nile/utils/get_nonce.py

```diff
@@ -0,0 +1,35 @@
+"""Retrieve nonce for a contract."""
+import logging
+import os
+import subprocess
+
+from nile.common import get_gateway
+
+GATEWAYS = get_gateway()
+
+
+def get_nonce(contract_address, network):
+    """Get the current nonce for contract address in a given network."""
+    nonce = get_nonce_without_log(contract_address, network)
+
+    logging.info(f"Current Nonce: {nonce}")
+
+    return nonce
+
+
+def get_nonce_without_log(contract_address, network):
+    """Get the current nonce without logging."""
+    # Starknet CLI requires an hex string for get nonce command
+    if type(contract_address) == int:
+        contract_address = hex(contract_address)
```

Comment:
```suggestion
    if not contract_address.startswith("0x"):
        contract_address = hex(int(contract_address))
```

After playing around with this, it seems `contract_address` is always passed as a string. The above suggestion is how I got it to work. Let me know what you think

Reply:
In case the method for some reason receives a int, I ended up with this from your suggestion:

```
if not str(contract_address).startswith("0x"):
    contract_address = hex(contract_address)
```

I will add a test specifically for this.

Reply:
Since `contract_address` is a string, I believe we'll have to convert it to an `int` before `hex`

> I will add a test specifically for this.

Great idea!

Reply:
Yes indeed, I will push with the update and the test.